lib/db/mysql.js

```diff
@@ -1082,5 +1084,46 @@ module.exports = function (log, error) {
       )
   }
 
+  function ipHmac(key, uid, addr) {
+    if (ip.isV4Format(addr)) {
+      addr = '::' + addr
+    }
+    addr = ip.toBuffer(addr)
+
+    var hmac = crypto.createHmac('sha256', key)
+    hmac.update(uid)
+    hmac.update(addr)
+
+    return hmac.digest()
+  }
+
+  var SECURITY_EVENT_NAMES = {
+    'account.create': 1,
+    'account.login': 2,
+    'account.reset': 3
+  }
+
+  var CREATE_SECURITY_EVENT = 'CALL createSecurityEvent_1(?, ?, ?, ?, ?)'
+  MySql.prototype.createSecurityEvent = function (data) {
+    var uid = data.uid
+    var tokenId = data.tokenId
+    var nameId = SECURITY_EVENT_NAMES[data.name]
+    var ipAddr = ipHmac(this.ipHmacKey, uid, data.ipAddr)
+    return this.write(CREATE_SECURITY_EVENT, [uid, tokenId, nameId, ipAddr, Date.now()])
+  }
+
+  var FETCH_SECURITY_EVENTS = 'CALL fetchSecurityEvents_1(?, ?)'
+  MySql.prototype.securityEvents = function (where) {
+    var uid = where.uid
+
+    var ipAddr = ipHmac(this.ipHmacKey, uid, where.ipAddr)
```

Comment:
It may be, but it will need more thought that I don't think should block this. In this case, we want the events that match the current `clientAddress`. We store the addresses as hmacs, so the auth-server cannot do the comparison itself unless we were to share the hmac key. Alternatively, we could return a boolean property `isCurrentAddr`, but that doesn't help the geo-profiling case.

The index is also on the pair `(uid, ipAddrHmac)`, so it'd be nice to keep the query that way too.